packages/backend/src/open_payments/quote/service.ts

```diff
@@ -0,0 +1,392 @@
+  if (options.receivingPayment) {
+    if (options.receivingAccount) {
+      return QuoteError.InvalidDestination
+    }
+    if (options.sendAmount && options.receiveAmount) {
+      return QuoteError.InvalidAmount
+    }
+  } else if (options.receivingAccount) {
+    if (options.sendAmount) {
+      if (options.receiveAmount || options.sendAmount.value <= BigInt(0)) {
+        return QuoteError.InvalidAmount
+      }
+    } else if (
+      !options.receiveAmount ||
+      options.receiveAmount.value <= BigInt(0)
+    ) {
+      return QuoteError.InvalidAmount
+    }
+  } else {
+    return QuoteError.InvalidDestination
+  }
+
+  const account = await deps.accountService.get(options.accountId)
+  if (!account) {
+    return QuoteError.UnknownAccount
+  }
+  if (options.sendAmount) {
+    if (
+      options.sendAmount.assetCode !== account.asset.code ||
+      options.sendAmount.assetScale !== account.asset.scale
+    ) {
+      return QuoteError.InvalidAmount
+    }
+  }
+
+  const plugin = deps.makeIlpPlugin({
+    sourceAccount: account,
+    unfulfillable: true
+  })
+
+  try {
+    await plugin.connect()
+
+    const destination = await resolveDestination(deps, options, plugin)
+
+    const receivingPaymentValue = destination.destinationPaymentDetails
+      ?.incomingAmount
+      ? destination.destinationPaymentDetails.incomingAmount.value -
+        destination.destinationPaymentDetails.receivedAmount.value
+      : undefined
+
+    const ilpQuote = await startQuote(deps, options, {
+      plugin,
+      destination,
+      sourceAsset: {
+        scale: account.asset.scale,
+        code: account.asset.code
+      }
+    })
+
+    return await Quote.transaction(deps.knex, async (trx) => {
+      assert.ok(destination.destinationPaymentDetails)
+      const quote = await Quote.query(trx)
+        .insertAndFetch({
+          accountId: options.accountId,
+          assetId: account.assetId,
+          receivingPayment: destination.destinationPaymentDetails.id,
+          sendAmount: {
+            value: ilpQuote.maxSourceAmount,
+            assetCode: account.asset.code,
+            assetScale: account.asset.scale
+          },
+          receiveAmount: {
+            value: ilpQuote.minDeliveryAmount,
+            assetCode: destination.destinationAsset.code,
+            assetScale: destination.destinationAsset.scale
+          },
+          // Cap at MAX_INT64 because of postgres type limits.
+          maxPacketAmount:
+            MAX_INT64 < ilpQuote.maxPacketAmount
+              ? MAX_INT64
+              : ilpQuote.maxPacketAmount,
+          minExchangeRate: ilpQuote.minExchangeRate,
+          lowEstimatedExchangeRate: ilpQuote.lowEstimatedExchangeRate,
+          highEstimatedExchangeRate: ilpQuote.highEstimatedExchangeRate,
+          completeReceivingPayment: !!(
+            options.receivingAccount && options.sendAmount
+          ),
+          // Patch using createdAt below
+          expiresAt: new Date()
```

Comment:
If this value isn't going to be used, maybe give it a more obviously-invalid value (e.g. `new Date(0)`).